modules/gcp/storage.go

```diff
@@ -0,0 +1,71 @@
+package gcp
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/gruntwork-io/terratest/modules/logger"
+)
+
+// CreateStorageBucket creates a Google Cloud bucket with the given BucketAttrs. Note that Google Storage bucket names must be globally unique.
+func CreateStorageBucket(t *testing.T, projectId string, name string, attr *storage.BucketAttrs) {
+	err := CreateStorageBucketE(t, projectId, name, attr)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// CreateStorageBucketE creates a Google Cloud bucket with the given BucketAttrs. Note that Google Storage bucket names must be globally unique.
+func CreateStorageBucketE(t *testing.T, projectId string, name string, attr *storage.BucketAttrs) error {
+	logger.Logf(t, "Creating bucket %s", name)
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
```

Comment:
`log.Fatalf`, believe it or not, calls `os.Exit`. Instead, just return the `err` here so the user can decide what to do with it. Same with all usages below.